File: 0908-middle-of-the-linked-list/0908-middle-of-the-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        vector<ListNode*> linked;
        
        ListNode* headPtr = head;

        while (headPtr != nullptr) {
            linked.push_back(headPtr);
            headPtr = headPtr->next;
        }

        if (linked.size()) {
            return linked[linked.size()/2];
        }

        return nullptr;
        
    }
};

File: 0908-middle-of-the-linked-list/0908-middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head == None or head.next == None:
            return head
        fast: ListNode = head.next
        slow: ListNode = head
        count: int = 0

        while fast != None:
            fast = fast.next
            if count % 2 == 0:
                slow = slow.next
            count+=1

        return slow